api/proto/metrics/metrics.proto

```diff
@@ -24,6 +24,42 @@ message memoryMetric {
   double availPercentage = 5;
 }
 
+// App Memory Metric
+message AppMemoryMetric {
+    // total_MB - Memory allocated to app instance by EVE. In Steady State,
+    //  this is expected to be same as in App instance config. Eve should
+    //  report what it actually allocated / reserved instead of what is configured.
+    uint32 total_MB = 1;
+
+    // used_MB - Memory Used by app from total_MB Memory
+    //  used_MB <= total_MB
+    uint32 used_MB = 2;
+
+    // avail_MB - Available memory remaining out of Total for the App.
+    //  total_MB >= used_MB + avail_MB
+    uint32 avail_MB = 3;
```

Comment:
Whatever's the additional semantics here that doesn't make `avail_MB` redundant -- it needs to be explained in the comments. IOW, why isn't `avail_MB == total_MB - used_MB`

Reply:
Yeah.. was a bit conflicted about it.. 

In the controller Vs. UserAgent discussion, UserAgent guys wanted this field to be explicitly filled up by controller - they preferred not to do any interpretation.

Thought I would follow a similar logic with Eve.

Will remove this..